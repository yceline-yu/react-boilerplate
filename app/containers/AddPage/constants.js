/*
 * AddPageConstants
 */

export const CHANGE_STRING = 'boilerplate/AddPage/CHANGE_STRING';
export const SUBMIT_STRING = 'boilerplate/AddPage/SUBMIT_STRING';
export const SUBMIT_ERROR = 'boilerplate/AddPage/SUBMIT_ERROR';
