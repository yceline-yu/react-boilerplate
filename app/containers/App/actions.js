/*
 * App Actions
 *
 */

import {
  LOAD_STRINGS,
  LOAD_STRINGS_SUCCESS,
  LOAD_STRINGS_ERROR,
} from './constants';

/**
 * Load the strings, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_STRINGS
 */
export function loadStrings() {
  return {
    type: LOAD_STRINGS,
  };
}

/**
 * Dispatched when the strings are loaded by the request saga
 *
 * @param  {object} strings The strings from server
 * @return {object} An action object with a type of LOAD_STRINGS_SUCCESS passing the strings
 */
export function stringsLoaded(strings) {
  return {
    type: LOAD_STRINGS_SUCCESS,
    strings,
  };
}

/**
 * Dispatched when loading the strings fails
 *
 * @param  {object} error The error
 *
 * @return {object} An action object with a type of LOAD_STRINGS_ERROR passing the error
 */
export function stringsLoadingError(error) {
  return {
    type: LOAD_STRINGS_ERROR,
    error,
  };
}
