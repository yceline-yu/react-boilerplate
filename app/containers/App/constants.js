/*
 * AppConstants
 *
 */

export const LOAD_STRINGS = 'boilerplate/App/LOAD_STRINGS';
export const LOAD_STRINGS_SUCCESS = 'boilerplate/App/LOAD_STRINGS_SUCCESS';
export const LOAD_STRINGS_ERROR = 'boilerplate/App/LOAD_STRINGS_ERROR';
