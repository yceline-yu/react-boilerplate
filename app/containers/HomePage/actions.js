// /*
//  * HomePage Actions
//  *
//  * Actions change things in your application
//  * Since this boilerplate uses a uni-directional data flow, specifically redux,
//  * we have these actions which are the only way your application interacts with
//  * your application state. This guarantees that your state is up to date and nobody
//  * messes it up weirdly somewhere.
//  *
//  * To add a new Action:
//  * 1) Import your constant
//  * 2) Add a function like this:
//  *    export function yourAction(var) {
//  *        return { type: YOUR_ACTION_CONSTANT, var: var }
//  *    }
//  */

// import {
//   LOAD_STRINGS,
//   LOAD_STRINGS_SUCCESS,
//   LOAD_STRINGS_ERROR,
// } from './constants';

// /**
//  * Load the strings, this action starts the request saga
//  *
//  * @return {object} An action object with a type of LOAD_STRINGS
//  */
// export function loadStrings() {
//   return {
//     type: LOAD_STRINGS,
//   };
// }

// /**
//  * Dispatched when the strings are loaded by the request saga
//  *
//  *
//  * @return {object}      An action object with a type of LOAD_STRINGS_SUCCESS passing the strings
//  */
// export function stringsLoaded(strings) {
//   return {
//     type: LOAD_STRINGS_SUCCESS,
//     strings,
//   };
// }

// /**
//  * Dispatched when loading the strings fails
//  *
//  * @param  {object} error The error
//  *
//  * @return {object}       An action object with a type of LOAD_STRINGS_ERROR passing the error
//  */
// export function stringsLoadingError(error) {
//   return {
//     type: LOAD_STRINGS_ERROR,
//     error,
//   };
// }
