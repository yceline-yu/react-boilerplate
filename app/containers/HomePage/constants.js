// /*
//  * AppConstants
//  * Each action has a corresponding type, which the reducer knows and picks up on.
//  * To avoid weird typos between the reducer and the actions, we save them as
//  * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
//  * reducers accidentally picking up actions they shouldn't.
//  *
//  * Follow this format:
//  * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
//  */

// export const LOAD_STRINGS = 'boilerplate/HomePage/LOAD_STRINGS';
// export const LOAD_STRINGS_SUCCESS = 'boilerplate/HomePage/LOAD_STRINGS_SUCCESS';
// export const LOAD_STRINGS_ERROR = 'boilerplate/HomePage/LOAD_STRINGS_ERROR';
