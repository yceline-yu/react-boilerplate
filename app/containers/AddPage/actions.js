/*
 * AddPage Actions
 *
 */

import { CHANGE_STRING, SUBMIT_STRING, SUBMIT_ERROR } from './constants';

/**
 * Changes the input field of the form
 *
 * @param  {string} string The new text of the input field
 *
 * @return {object} An action object with a type of CHANGE_STRING
 */
export function changeString(string) {
  return {
    type: CHANGE_STRING,
    string,
  };
}

/**
 * Changes status of the form
 *
 * @param  {string} submit The final version of the input field
 *
 * @return {object} An action object with a type of SUBMIT_STRING
 */
export function submitString(submit) {
  return {
    type: SUBMIT_STRING,
    submit,
  };
}

/**
 * Dispatched when posting fails
 *
 * @param  {object} error The error
 *
 * @return {object} An action object with a type of SUBMIT_ERROR passing the error
 */
export function submitError(error) {
  return {
    type: SUBMIT_ERROR,
    error,
  };
}
