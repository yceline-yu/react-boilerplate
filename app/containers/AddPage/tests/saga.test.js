/**
 * Tests for AddPage sagas
 */

import { put, takeLatest } from 'redux-saga/effects';

import { SUBMIT_STRING } from '../constants';
import { submitError } from '../actions';

import serverData, { postString } from '../saga';

/* eslint-disable redux-saga/yield-effects */
describe('getStrings Saga', () => {
  let postStringsGenerator;

  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    const fixture = { submit: 'string' };
    postStringsGenerator = postString(fixture);
    const callDescriptor = postStringsGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the stringsLoaded action if it requests the data successfully', () => {
    const response = ['test', 'test'];
    const putDescriptor = postStringsGenerator.next(response).value;
    expect(putDescriptor).toEqual(undefined);
  });

  it('should call the submitError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = postStringsGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(submitError(response)));
  });
});

describe('serverDataSaga Saga', () => {
  const serverDataSaga = serverData();

  it('should start task to watch for LOAD_STRINGS action', () => {
    const takeLatestDescriptor = serverDataSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(SUBMIT_STRING, postString));
  });
});
