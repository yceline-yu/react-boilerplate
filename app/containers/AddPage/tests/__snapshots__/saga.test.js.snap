// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStrings Saga should call the submitError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3001/strings",
      Object {
        "body": "{\\"string\\":\\"string\\"}",
        "headers": Object {
          "Accept": "application/json",
          "Content-Type": "application/json",
        },
        "method": "POST",
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getStrings Saga should dispatch the stringsLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3001/strings",
      Object {
        "body": "{\\"string\\":\\"string\\"}",
        "headers": Object {
          "Accept": "application/json",
          "Content-Type": "application/json",
        },
        "method": "POST",
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
